lib/solr.js

```diff
@@ -827,120 +884,60 @@ Client.prototype.ping = function (callback) {
   return this.get(this.ADMIN_PING_HANDLER, callback);
 };
 
-/**
- * HTTP POST request. Send update commands to the Solr server using form encoding (e.g. search)
- *
- * @param {Object} params
- * @param {String} params.host - IP address or host address of the Solr server
- * @param {Number|String} params.port - port of the Solr server
- * @param {String} params.core - name of the Solr core requested
- * @param {String} params.authorization - value of the authorization header
- * @param {String} params.fullPath - full path of the request
- * @param {String} params.params - form params
- * @param {Boolean} params.secure -
- * @param {Boolean} params.bigint -
- * @param {http.Agent} [params.agent] -
- * @param {Object} params.request - request options
- * @param {Function} callback(err,obj) - a function executed when the Solr server responds or an error occurs
- * @param {Error} callback().err
- * @param {Object} callback().obj - JSON response sent by the Solr server deserialized
- *
- * @return {http.ClientRequest}
- * @api private
- */
+Client.prototype.createSchemaField = function (fieldName, fieldType, cb) {
+  const json = JSON.stringify({
+    'add-field': {
+      name: fieldName,
+      type: fieldType,
+      multiValued: false,
+      stored: true,
+    },
+  });
 
-function postForm(params, callback) {
-  const headers = {
-    'content-type': 'application/x-www-form-urlencoded; charset=utf-8',
-    'content-length': Buffer.byteLength(params.params),
-    accept: 'application/json; charset=utf-8',
+  const callback = (err, data) => {
+    if (err) {
+      // TODO: We should handle this in a more robust way in the future, but
+      //   there is a difference between default setup in Solr 5 and Solr 8, so
+      //   some fields already exist in Solr 8. Hence if that's the case, we
+      //   just ignore that.
+      console.warn(err.message);
+    }
+    cb(undefined, data);
   };
-  if (params.authorization) {
-    headers['authorization'] = params.authorization;
-  }
-  let options = {
-    host: params.host,
-    port: params.port,
+
+  const requestOptions = {
+    host: this.options.host,
+    port: this.options.port,
     method: 'POST',
-    headers: headers,
-    path: params.fullPath,
-    family: params.ipVersion,
+    headers: {
+      'content-type': 'application/json; charset=utf-8',
+      'content-length': Buffer.byteLength(json),
+      accept: 'application/json; charset=utf-8',
+    },
+    path: 'http://localhost:8983/solr/testcore/schema',
```

Comment:
why does this have hardcoded path?

Reply:
It's part of the original code. I just moved the method, I did not write it. Currently this method is only used during tests, so the URL is always the same. We should fix that in a separate issue.

Reply:
Actually I can include that fix in this PR. Just a moment... :)